dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -1203,43 +1232,69 @@ BlockInputStreams DeltaMergeStore::read(const Context & db_context,
 {
     // Use the id from MPP/Coprocessor level as tracing_id
     auto dm_context = newDMContext(db_context, db_settings, tracing_id);
-
-    SegmentReadTasks tasks = getReadTasksByRanges(*dm_context, sorted_ranges, num_streams, read_segments);
+    // TODO(jinhelin): If keep order is required, disable read thread.
+    auto enable_read_thread = db_context.getSettingsRef().dt_enable_read_thread;
+    SegmentReadTasks tasks = getReadTasksByRanges(*dm_context, sorted_ranges, num_streams, read_segments, !enable_read_thread);
 
     auto tracing_logger = Logger::get(log->name(), dm_context->tracing_id);
     LOG_FMT_DEBUG(tracing_logger, "Read create segment snapshot done");
 
-    auto after_segment_read = [&](const DMContextPtr & dm_context_, const SegmentPtr & segment_) {
+    [[maybe_unused]] auto after_segment_read = [&](const DMContextPtr & dm_context_, const SegmentPtr & segment_) {
         // TODO: Update the tracing_id before checkSegmentUpdate?
         this->checkSegmentUpdate(dm_context_, segment_, ThreadType::Read);
     };
 
     GET_METRIC(tiflash_storage_read_tasks_count).Increment(tasks.size());
     size_t final_num_stream = std::max(1, std::min(num_streams, tasks.size()));
-    auto read_task_pool = std::make_shared<SegmentReadTaskPool>(std::move(tasks));
+    auto read_task_pool = std::make_shared<SegmentReadTaskPool>(
+        physical_table_id,
+        dm_context,
+        columns_to_read,
+        filter,
+        max_version,
+        expected_block_size,
+        false,
+        db_settings.dt_raw_filter_range,
```

Comment:
Since that info are both used by  `SegmentReadTaskPool` and  `DMSegmentThreadInputStream`, how about grouping that info into a struct and passing around?

Reply:
I think can use `SegmentReadTaskPool` to create inputsteam of segment for `DMSegmentThreadInputStream` and `DMSegmentThreadInputStream` is not necessarily to store these infomation.

I will refine the related code if I have time these two days or maybe later...